classwork/сlasswork2.1/main.cpp

```diff
@@ -0,0 +1,108 @@
+#include <iostream>
+#include <unordered_map>
+#include <fstream>
+#include <string>
+#include <print>
+#include <format>
+
+namespace {
+    constexpr const char* DEFAULT_FILE_NAME = "user_stats.txt";
+    constexpr const char* RESET_COMMAND = "bread";
+    constexpr const char* DELETE_COMMAND = "delete";
+}
+
+class UserStatistics {
+public:
+    explicit UserStatistics(const std::string& statisticsFilePath = DEFAULT_FILE_NAME)
+            : fileName(statisticsFilePath) {
+        loadFromFile();
+    }
+
+    ~UserStatistics() {
+        saveToFile();
+    }
+
+    bool incrementCounterForUser(const std::string& name, std::string& message) {
+        if (name == RESET_COMMAND) {
+            userStats.clear();
+            saveToFile();
+            message = "All history has been exterminated.";
+            return true;
+        }
+
+        if (userStats.find(name) == userStats.end()) {
+            userStats[name] = 1;
+            message = std::format("Welcome, {}!", name);
+            return true;
+        } else {
+            int count = ++userStats[name];
+            message = std::format("Hello again({}), {}!", count, name);
+            return true;
```

Comment:
you always return true for any branch in this function so such a return value is actually doesn't make sense.